lib/compress/index.js

```diff
@@ -5177,6 +5177,49 @@ merge(Compressor.prototype, {
             return true;
         }
 
+        function can_inject_args_values() {
+            var arg_vals_outer_refs = new Set();
+            var value_walker = new TreeWalker(function(node) {
+                if (node instanceof AST_Scope) {
```

Comment:
This will not find block scopes. Block scopes are the `block_scope` property of each block.

Reply:
I think that sucks, but it's because Terser was built from the ground up for ES5, with the idea that a scope is always a function. This is of course not true anymore :'(

Reply:
I must say, I was slightly confused about how block scopes were integrated in the compressor. However I did not check for it, because an argument has to be an expression. It can only really have an inner scope either when a lambda function is passed as an argument or some expression involving an IIFE, right? I could not think of a way it would directly contain a block scope. It can of course contain a block scope inside the lambda or IIFE, but I assumed all symbols refering to outer scopes would then be mentioned in the lambda/IIFE's enclosed property.

Please correct me if I'm wrong.

Reply:
Anyone? It would be nice if we can conclude if my thinking is correct. If it isn't I'll try to fix it, if it is this bug is solved.

Reply:
@j-verse from my end I think that this PR is good but I'm not sure that I actually get what did @fabiosantoscode mean.

My example input:

```js
var g = ["a"];

function problem(arg) {
    return g.indexOf(arg);
}

function unused(arg) {
    return problem(arg);
}

try {
    function a(arg) {
        return problem(arg);
    }

    function b(problem) {
        return g[problem];
    }

    function c(arg) {
        return b(a(arg));
    }

    block: {
        let g = null;
        let c = Math.round(Math.random() + 1);
        switch (c) {
            case 1: {
                let b = a("a");
                console.log(b);
                break;
            }
            case 2: {
                let c = b(a("a"));
                console.log(c);
                break;
            }
            default:
                console.log("c");
        }
    }

} catch (e) {
    console.log("Error!");
}
```

And Terser output:
```js
var g = ["a"];

function problem(arg) {
    return g.indexOf(arg);
}

try {
    function a(arg) {
        return problem(arg);
    }

    function b(problem) {
        return g[problem];
    }

    switch (Math.round(Math.random() + 1)) {
        case 1: {
            let b = a("a");
            console.log(b);
            break;
        }
        case 2: {
            let c = b(a("a"));
            console.log(c);
            break;
        }
        default:
            console.log("c");
    }
} catch (e) {
    console.log("Error!");
}
```